### PR TITLE
feat(`monitoringc`): add verification of client for IBC handshake

### PR DESCRIPTION
**Pull request by lubtd**
Scaffold map `ProviderClientID` to register the official client ID of a provider chain from the launch ID once the IBC connection is established
Scaffold map `LaunchIDFromVerifiedClientID` to retrieve the associated launch ID for a verified client and check if a client is verified
Implement handshake verification in `module_ibc.go`

TODO

- [x] mocking capability for IBC modules
- [x] test for `handshake.go`

### CODE REVIEW

File: proto/monitoringc/provider_client_id.proto [hunk 1]
```diff
@@ -0,0 +1,11 @@
+syntax = "proto3";
+package tendermint.spn.monitoringc;
+
+option go_package = "github.com/tendermint/spn/x/monitoringc/types";
+
+message ProviderClientID {
+  uint64 launchID = 1; 
+  string clientID = 2; 
+
```
Comment (Pantani) [inline proto/monitoringc/provider_client_id.proto:9]:
```suggestion
```

File: x/monitoringc/client/cli/query_launch_id_from_verified_client_id.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/spn/x/monitoringc/types"
+)
+
+func CmdListLaunchIDFromVerifiedClientID() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list-launch-id-from-verified-client-id",
+		Short: "list all LaunchIDFromVerifiedClientID",
```
Comment (Pantani) [inline x/monitoringc/client/cli/query_launch_id_from_verified_client_id.go:15]:
```suggestion
		Short: "list all LaunchIDFromVerifiedClientID",
		Args:  cobra.NoArgs,
```

File: x/monitoringc/module.go [hunk 3]
```diff
@@ -4,7 +4,7 @@ import (
 	"encoding/json"
 	"fmt"
 
-	// this line is used by starport scaffolding # 1
+	"context"
```
Comment (Pantani) [inline x/monitoringc/module.go:7]:
```suggestion
	"context"
	"encoding/json"
	"fmt"

```

**Review by Pantani — CHANGES_REQUESTED** [review]

File: x/monitoringc/client/cli/query.go [hunk 4]
```diff
@@ -27,6 +27,10 @@ func GetQueryCmd(queryRoute string) *cobra.Command {
 	cmd.AddCommand(CmdQueryParams())
 	cmd.AddCommand(CmdVerifiedClientIds())
 
+	cmd.AddCommand(CmdListProviderClientID())
+	cmd.AddCommand(CmdShowProviderClientID())
+	cmd.AddCommand(CmdListLaunchIDFromVerifiedClientID())
+	cmd.AddCommand(CmdShowLaunchIDFromVerifiedClientID())
```
[the remark this answers is not in the document]
Comment (lubtd) [reply x/monitoringc/client/cli/query.go:33]: Params as last

File: x/monitoringc/module.go [hunk 5]
```diff
@@ -78,7 +77,7 @@ func (AppModuleBasic) RegisterRESTRoutes(clientCtx client.Context, rtr *mux.Rout
 
 // RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the module.
 func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
-	// this line is used by starport scaffolding # 2
+	types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx))
```
Comment (Pantani) [inline x/monitoringc/module.go:80]:
```suggestion
	types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx)) // nolint
```

**Review by Pantani — APPROVED** [review]

**Pull request merged**